src/main/resources/db/migration/V2017_10_16_1600__Add_external_reference_to_claim.sql

```diff
@@ -0,0 +1,2 @@
+ALTER TABLE claim
+  ADD COLUMN external_reference TEXT;
```

Comment:
I'm not sure if this belongs here,
we usually keep all user entered data on the claim.
You can query on json fine..

Completely up for discussion though 👍

Reply:
as `claim` table should be in fact `case` and `external_reference` is related to everything for this case IMHO separate column is a good idea.

Reply:
I thought of implementing search on json objects, but my concern was the performance for search query on json. Had discussion with @dharmendrak  as well yesterday. I am happy to change to json search if that has better performance. Open for better options here.

Reply:
I would say performance is not a concern here...

Reply:
separate column is good idea, as  all our searches are happening on claim elements not on claim data (json blob). we can write a index on external reference number and reference number to increase the performance of database.  ExternalId is also part of claim data, however it is extracted and stored separately. I support for separate column

Reply:
we can put an index on the json data too.
Performance is not a concern here, as T2 will be on CCD before that matters and then we need to solve search properly.
I still don't think it should go on its own column as it violates our current principal of user entered data on the claim data

Reply:
I agree with Tim that once we're on CCD we require to redo searches. Also external-id doesn't uniquely identify a case like reference because user can enter duplicate values. I suggest having an index on jsonB column on external-Id and hold onto our structure as is.

Reply:
@singh-surjeet implemented search on json claim data.